client/op.go

```diff
@@ -0,0 +1,398 @@
+package client
+
+import xlineapi "github.com/xline-kv/go-xline/api/xline"
+
+// OpOption configures Operations like Get, Put, Delete.
+type OpOption func(*Op)
+
+// Op represents an Operation that kv can execute.
+type Op struct {
+	t   opType
+	key []byte
+	end []byte
+
+	// for range
+	limit             int64
+	sort              *SortOption
+	serializable      bool
+	keysOnly          bool
+	cntOnly           bool
+	minModRev         int64
+	maxModRev         int64
+	minCreateRev      int64
+	maxCreateRev      int64
+	isOptsWithFromKey bool
+	isOptsWithPrefix  bool
+
+	// for kv
+	val      []byte
+	leaseID  int64
+	prevKV   bool
+	ignVal   bool
+	ignLease bool
+
+	// txn
+	cmps    []Cmp
+	thenOps []Op
+	elseOps []Op
+
+	// for compact
+	physical bool
+
+	// for range, compact
+	rev int64
+}
+
+type opType int
+
+const (
+	// A default Op has opType 0, which is invalid.
+	opRange opType = iota + 1
+	opPut
+	opDeleteRange
+	opTxn
+	opCompact
+)
+
+func NewOp() *Op {
+	return &Op{key: []byte("")}
+}
+
+func (op *Op) applyOpts(opts []OpOption) {
+	for _, opt := range opts {
+		opt(op)
+	}
+}
+
+// OpGet returns "get" operation based on given key and operation options.
+func OpGet(key string, opts ...OpOption) Op {
+	// WithPrefix and WithFromKey are not supported together
+	if IsOptsWithPrefix(opts) && IsOptsWithFromKey(opts) {
+		panic("`WithPrefix` and `WithFromKey` cannot be set at the same time, choose one")
+	}
+	op := Op{t: opRange, key: []byte(key)}
+	op.applyOpts(opts)
+	return op
+}
+
+// OpPut returns "put" operation based on given key-value and operation options.
+func OpPut(key, val string, opts ...OpOption) Op {
+	op := Op{t: opPut, key: []byte(key), val: []byte(val)}
+	op.applyOpts(opts)
+	switch {
+	case op.end != nil:
+		panic("unexpected range in put")
+	case op.limit != 0:
+		panic("unexpected limit in put")
+	case op.rev != 0:
+		panic("unexpected revision in put")
+	case op.sort != nil:
+		panic("unexpected sort in put")
+	case op.serializable:
+		panic("unexpected serializable in put")
+	case op.cntOnly:
+		panic("unexpected countOnly in put")
+	case op.minModRev != 0, op.maxModRev != 0:
+		panic("unexpected mod revision filter in put")
+	case op.minCreateRev != 0, op.maxCreateRev != 0:
+		panic("unexpected create revision filter in put")
+		// case op.filterDelete, op.filterPut:
```

Comment:
Why do you comment on this code? If we don't use it, please remove it.

Reply:
Deleted.